addon/components/bs-form/element.d.ts

```diff
@@ -0,0 +1,41 @@
+import Component from '@glimmer/component';
+
+interface BsFormElementSignature<T> {
+  Element: HTMLDivElement;
+  Args: {
+    controlType?: 'text' | 'checkbox' | 'radio' | 'switch' | 'textarea' | string;
```

Comment:
Would be awesome if we could detect the registered control components. Not sure how that would work. But I guess it would be only needed as interim solution anyways. I assume we are going to deprecate that feature when [first-class component templates](https://rfcs.emberjs.com/id/0779-first-class-component-templates/) are available.

Reply:
as mentioned elsewhere, this should be just an arg for <input type={{controlType}} html attribute, so string should be enough.

Reply:
See https://github.com/ember-bootstrap/ember-bootstrap/pull/1861#discussion_r1097477232